comptests/s3/test_user_policy_crud.py

```diff
@@ -0,0 +1,412 @@
+                if status[0] != HTTPStatus.OK:
+                    delete_failed.append(usr)
+                else:
+                    delete_success.append(usr)
+            # pylint: disable=broad-except
+            except BaseException as err:
+                self.log.warning("Ignoring %s while deleting user: %s", err, usr)
+        for usr in delete_success:
+            self.created_users.remove(usr)
+        self.log.info("User delete success list %s", delete_success)
+        self.log.info("User delete failed list %s", delete_failed)
+
+    @pytest.mark.api_user_ops
+    @pytest.mark.tags('TEST-41407')
+    def test_41407(self):
+        """ Test to DeleteUserPolicy to the Valid user. """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("Test to DeleteUserPolicy to the user.")
+        self.log.info("Step1 : Creating IAM user.")
+        user_name = f"{self.user_name_prefix}{str(time.perf_counter_ns()).replace('.', '_')}"
+        email=f"{user_name}@seagate.com"
+        user_params1 = {
+            'display-name': user_name,
+            'email' : email,
+            'uid' : user_name
+        }
+        self.log.info("Step 2: Creating a new IAM user with name %s", str(user_name))
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params1))
+        self.log.info(user_info)
+        self.log.info("Step 3: Verifying that new IAM user is created successfully")
+        assert status == HTTPStatus.OK, "Not able to create user. Test Failed"
+        self.log.info("Step 4: IAM user is created successfully")
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Deny",
+                    "Action": "s3:PutObject",
+                    "Resource": "arn:aws:s3:::test1/*"
+                }
+            ]
+        }
+        self.log.info("Step 5: Applying user policy to the user")
+        self.log.info(policy)
+        policy_document = json.dumps(policy)
+        user_params2 = {
+            'UserName' : user_name,
+            'PolicyName' : 'Policy1',
+            'PolicyDocument' : policy_document,
+            'Action' : 'PutUserPolicy',
+            'format' : 'json'
+        }
+        loop = asyncio.get_event_loop()
+        status = loop.run_until_complete(self.obj.put_user_policy(user_params2))
+        self.log.info("Step 6: verifying Applied user policy to the user")
+        assert status[0] == HTTPStatus.OK, "Not able to apply policy"
+        self.log.info("Step 7:  Applied user policy Successfully")
+        user_params3 = {
+            'UserName' : user_name,
+            'PolicyName' : 'Policy1',
+            'Action' : 'DeleteUserPolicy',
+            'format' : 'json'
+        }
+        loop = asyncio.get_event_loop()
```

Comment:
Extract out the commonly used functions like create user and create policy into generic functions and  push it down to test lib layer.